### PR TITLE
update vault instructions to use mysql-innodb

### LINKED ISSUE #609: Vault example hasn't been updated for focal+
Related: https://bugs.launchpad.net/charmed-kubernetes-bundles/+bug/1946290

The Vault overlay examples still use the percona-cluster charm which is not available for focal. Would be nice to have an up-to-date example with mysql-innodb-cluster and mysql-router.

https://github.com/charmed-kubernetes/kubernetes-docs/blob/fb7c58e560206a03df0ea7f952561fe234d15a5b/pages/k8s/using-vault.md?plain=1#L43-L45

https://github.com/charmed-kubernetes/kubernetes-docs/search?q=percona-cluster

### PR DESCRIPTION
**Pull request by kwmonroe**
Update vault instructions to use the current recommended database and best practices from https://charmhub.io/vault.

Fixes #609 and [LP:1946290](https://bugs.launchpad.net/charmed-kubernetes-bundles/+bug/1946290)

### CODE REVIEW

File: pages/k8s/encryption-at-rest.md [hunk 1]
```diff
@@ -18,45 +18,84 @@ needed by a cluster, such as usernames and passwords, encryption keys, etc.
...
+juju run-action vault/0 authorize-charm token={charm token} --wait
 ```
 
-Once **Vault** comes up (and any time it is rebooted), you will need to [unseal][]
-it.
+<div class="p-notification--information">
+  <div markdown="1" class="p-notification__content">
+    <p class="p-notification__message">It is <strong><em>critical </em></strong> that you save all five unseal keys as well as the root token.  If the <strong>Vault</strong> unit is ever rebooted, you will have to repeat the unseal steps (but not the init step) before the CA can become functional again.</p>
+  </div>
+</div>
```
Comment (addyess) [inline pages/k8s/encryption-at-rest.md:95]: I **LOVE** this.  😍

Comment (addyess) [reply pages/k8s/encryption-at-rest.md:95]: you can't emphasize this enough

**Review by addyess — APPROVED** [review]
LGTM

**Review by evilnick — APPROVED** [review]
:cake:

**Pull request merged**